### PR TITLE
Add S3Handle using minio 5.0.0

### PR DESCRIPTION
**Pull request by joshmoore**
URLs starting with `s3://` will now be passed to the
S3Handle. `./showinfo s3://bucket/path/file.tiff` then
opens the file as expected.

 * [ ] introduce caching
 * [ ] add login information (??)
 * [ ] add documentation

### CODE REVIEW

**Comment by joshmoore** [comment]
Caching looks like this:

```
        s3Client.getObject(
                "bucket",
                "/browse/formatted/20180823/iss_breast/dots-fov_000-Z0-H0-C0.tiff",
                "/tmp/cache-dir");
```

**Comment by joshmoore** [comment]
With the latest 2 commits, we can now read from our own minio server (as opposed to AWS) but there seems to be some corruption:

```
loci.formats.in.TiffReader.initFile(s3://local-bucket.localhost:9000/hybridization-fov_000_Z0_T0_C0.ome.tiff)
Reading IFDs
Error reading IFD type at: 206

loci.common.enumeration.EnumException: Unable to find IFDType with code: 0
	at loci.formats.tiff.IFDType.get(IFDType.java:90) ~[bioformats_package.jar:5.10.0-SNAPSHOT]
	at loci.formats.tiff.TiffParser.readTiffIFDEntry(TiffParser.java:1320) [bioformats_package.jar:5.10.0-SNAPSHOT]
	at loci.formats.tiff.TiffParser.getIFD(TiffParser.java:440) [bioformats_package.jar:5.10.0-SNAPSHOT]
	at loci.formats.tiff.TiffParser.getMainIFDs(TiffParser.java:224) [bioformats_package.jar:5.10.0-SNAPSHOT]
	at loci.formats.in.MinimalTiffReader.initFile(MinimalTiffReader.java:459) [bioformats_package.jar:5.10.0-SNAPSHOT]
	at loci.formats.in.BaseTiffReader.initFile(BaseTiffReader.java:583) [bioformats_package.jar:5.10.0-SNAPSHOT]
	at loci.formats.FormatReader.setId(FormatReader.java:1397) [bioformats_package.jar:5.10.0-SNAPSHOT]
	at loci.formats.ReaderWrapper.setId(ReaderWrapper.java:650) [bioformats_package.jar:5.10.0-SNAPSHOT]
	at loci.formats.tools.ImageInfo.testRead(ImageInfo.java:1034) [bioformats_package.jar:5.10.0-SNAPSHOT]
	at loci.formats.tools.ImageInfo.main(ImageInfo.java:1120) [bioformats_package.jar:5.10.0-SNAPSHOT]
```

screen shot 2018-09-27 at 00 16 12

**Comment by joshmoore** [comment]
Ah, minio is compiled for JDK8+. cc: @sbesson @melissalinkert @dgault @manics

File: src/main/java/loci/common/Location.java [hunk 1]
```diff
@@ -145,7 +129,40 @@ public Location(File file) {
...
    */
   public Location(String parent, String child) {
-    this(parent + File.separator + child);
+    LOGGER.trace("Location({}, {})", parent, child);
+
+    String mapped = null;
+    String pathname = null;
+
+    // First handle possible URIs
+    if (child.contains("://")) {
```
Comment (sbesson) [inline src/main/java/loci/common/Location.java]: Under some conditions this seem to trigger a NPE - see https://web-proxy.openmicroscopy.org/east-ci/job/BIOFORMATS-test-folder/35481/console

Comment (joshmoore) [reply src/main/java/loci/common/Location.java]: Ha. Ok. That's interesting.

https://github.com/snoopycrimecop/bioformats/blob/develop/merge/daily/components/formats-gpl/src/loci/formats/in/InCellReader.java#L863 does

```
      else if (qName.equals("Image")) {
        String file = attributes.getValue("filename");
        String thumb = attributes.getValue("thumbnail");
        Location current = new Location(currentId).getAbsoluteFile();

        Location imageFile = new Location(current.getParentFile(), file);
        currentImageFile = imageFile.getAbsolutePath();
        currentThumbnail =
          new Location(current.getParentFile(), thumb).getAbsolutePath();
      }
```

and https://github.com/ome/ome-common-java/blob/master/src/main/java/loci/common/Location.java#L148 does:

```
    this(parent + File.separator + child);
```

so I assume the current behavior is to return a Location ending in "null". I'll add the test for null here, but this probably could reuse further review.
